### DISCUSSION TITLE
is it possible to set two factors as strata?

### DISCUSSION THREAD

**Discussion by Biofarmer** [vegandevs/vegan#689] [Q&A]
Hi developer,

May I ask a question about the function of adonis2?
I want to set "subjects" and "locations" as random factors in adonis2, like repeated measured PRMANOVA? How to realize it?

Thank you very much.
Best,
Wang

**Comment by gavinsimpson** [comment]
These are fixed effects so you add them as you would any other factor.

Right now, we do not compute the correct denominator in the "ANOVA" table as we use the residual sum of squares for the denominator for normal and random factors.

You do need to set `subjects` and `locations` as restrictions in the permutation design. How you do this will depend on what hypothesis you want to test?
